### DISCUSSION TITLE
More spectrogram help: audiovisual project

### DISCUSSION THREAD

**Discussion by OndrejBocek** [audacity/audacity#4990] [Q&A]
Hey everyone,

So, I'm asking for some more help with Audacity spectrogram view. This time, it is a bit more complex. What is going on is essentially this: in my AV project, my goal is to take a source wav file, extract certain frequencies and thereafter generate a sinewave and a spectrogram based on these frequencies - all this with adjustable variables. So far I've extracted **formants** from the source audio file in PRAAT. I've put together a python code that generates a **sinewave** based on the formants (which by themselves are just frequency values in time). I've allowed myself several variables I can use later for adjustments of the wave. Either way, the wave is saved as a wav file.

The next step I struggle with. The goal is to generate a spectrogram of the wav file. Of course, I can do this in python with some of the available libraries. However, the thing is I absolutely _love_ Audacity spectrogram view ever since my linguistics undergrad. It is visually gorgeous - which does matter for the project. Unfortunately, going into Audacity and selecting spectrogram view for the wav file (and then screenshotting) is out of question - the wav file will be edited constantly in real time and I _need_ this to work in code, not Audacity. The shifting spectrogram (shifting because I'll edit the sinewave variables) will be projected afterwards. 

Trust me, this could be absolutely beautiful. I'm planning to change the sinewave variables with the grandMA light designing software which enables cool effects and an overall control of numerical variables. What I need from audacity is the **spectrogram code** which will pick up the wav file and generate the spectrogram as an img file. It is also desirable to be able to edit the spectrogram settings in real time.

Please let me know if you'd be willing to help me out a bit. I am not asking anyone to do all the work for me. I'd appreciate any amount of help, as for now I'm more or less completely lost - yet I just can't give up on the beautiful Audacity spectrograms.  I'm absolutely fine with crediting the Audacity software and anyone who helps once the project is done.

Thank you.

EDIT: If you can tell what I want is unachievable with Audacity, that is also a very important piece of information,
EDIT2: Would all this be achievable through Audacity scripting?

**Comment by spur19** [comment]
Super interested in doing the same thing. Let me know if you figured this out.
